packages/datadog-instrumentations/src/graphql.js

```diff
@@ -235,14 +235,28 @@ function callInAsyncScope (fn, thisArg, args, abortController, cb) {
   try {
     const result = fn.apply(thisArg, args)
     if (result && typeof result.then === 'function') {
-      // bind callback to this scope
-      result.then(
-        res => cb(null, res),
-        err => cb(err)
+      let promise = result
+
+      // Mongoose queries are thenable, but not true promises. They can't be
+      // resolved multiple times. `.exec()` returns a true promise. We can
+      // identify a mongoose query by the presence of an `exec` function and
+      // its constructor name.
+      if (typeof result.exec === 'function' && result.constructor.name === 'Query') {
+        promise = result.exec()
```

Comment:
Does exec() have side effects? Can it be called multiple times?

Reply:
Fixed to wrap the query object instead of calling `.exec()`